packages/client-core/src/bodySerializer.ts

```diff
@@ -4,24 +4,16 @@ import type {
   SerializerOptions,
 } from './pathSerializer';
 
-export type QuerySerializer = (query: Record<string, unknown>) => string;
-
 export type BodySerializer = (body: any) => any;
 
+export type QuerySerializer = (query: Record<string, unknown>) => string;
+
 export interface QuerySerializerOptions {
   allowReserved?: boolean;
   array?: SerializerOptions<ArrayStyle>;
   object?: SerializerOptions<ObjectStyle>;
 }
 
-const serializeFormDataPair = (data: FormData, key: string, value: unknown) => {
-  if (typeof value === 'string' || value instanceof Blob) {
-    data.append(key, value);
-  } else {
-    data.append(key, JSON.stringify(value));
```

Comment:
It might be nice to warn when stringifying a non-JSON type. I ran into issues initially when passing a `FileList` to a form, expecting it to be serialized as a blob array. Instead, it came out as `{}`.

Reply:
Honestly I think that's more on us to handle properly